core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -28,18 +28,35 @@ oppia.directive('collectionNodeCreator', [
         'validatorsService', 'CollectionEditorStateService',
         'CollectionLinearizerService', 'CollectionUpdateService',
         'CollectionNodeObjectFactory', 'ExplorationSummaryBackendApiService',
-        'siteAnalyticsService',
+        'SearchExplorationsBackendApiService', 'siteAnalyticsService',
         function(
             $scope, $http, $window, $filter, alertsService,
             validatorsService, CollectionEditorStateService,
             CollectionLinearizerService, CollectionUpdateService,
             CollectionNodeObjectFactory, ExplorationSummaryBackendApiService,
-            siteAnalyticsService) {
+            SearchExplorationsBackendApiService, siteAnalyticsService) {
           $scope.collection = CollectionEditorStateService.getCollection();
           $scope.newExplorationId = '';
           $scope.newExplorationTitle = '';
           var CREATE_NEW_EXPLORATION_URL_TEMPLATE = '/create/<exploration_id>';
 
+          $scope.fetchExplorationsMetadata = function(searchQuery) {
+            if (/^[a-zA-Z0-9- ]*$/.test(searchQuery)) {
+              return SearchExplorationsBackendApiService.getExplorations(
+                searchQuery).then(
+              function(explorationMetadataObject) {
```

Comment:
Indentation:

```
              return SearchExplorationsBackendApiService.getExplorations(
                searchQuery
              ).then(function(explorationMetadataObject) {
                ...
              }, function(...) {
                ...
              });
```

I think you can also call this explorationMetadataBackendDict to ensure there's clarity that it uses backend naming conventions for its keys.